pkg/thanos/statefulset.go

```diff
@@ -294,7 +296,26 @@ func makeStatefulSetSpec(tr *monitoringv1.ThanosRuler, config Config, ruleConfig
 	if tr.Spec.AlertQueryURL != "" {
 		trCLIArgs = append(trCLIArgs, monitoringv1.Argument{Name: "alert.query-url", Value: tr.Spec.AlertQueryURL})
 	}
-
+	if len(tr.Spec.Version) > 0 {
+		version, err := semver.ParseTolerant(tr.Spec.Version)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse thanos ruler version")
+		}
```

Comment:
The if and parsing the version is not needed here, we already do it at the very beginning of `makeStatefulSetSpec`